vortex-expr/src/analysis.rs

```diff
@@ -35,15 +35,20 @@ pub trait AnalysisExpr {
         None
     }
 
-    /// If an expression is returned, its value is an upper bound on the value of `expr`.
+    /// An expression for the upper non-null bound of this expression, if available.
     ///
-    /// We may return `None` for values which have no upper bound or values for which knowing the
-    /// upper bound is difficult.
+    /// This function returns None if there is no upper bound or it is difficult to compute.
```

Comment:
Sorry nitpick, but I think the point is *unknown* (due to any reason, including not implemented)
```suggestion
    /// This function returns None if there the upper bound is unknown (non existent or hard to compute).
```

Reply:
Sure, I suppose undefined (max of an empty list) is just another kind of unknown.